### DISCUSSION TITLE
Jinx marking "shouldn" as misspelling in comment

### DISCUSSION THREAD

**Discussion by ashton314** [minad/jinx#170] [Q&A]
I'm working in `elixir-mode`, and Jinx is highlighting the chunk "shouldn" in "shouldn't" as misspelled. This doesn't happen `emacs-lisp-mode` or `haskell-mode`, so it's probably a problem in  `elixir-mode`. I would love a pointer of where to look to see if I can fix this.

Here's an `elpaca-test` expression that highlights the error: (I dunno if you're using Elpaca or not. If you are, you should be able to drop the below code in a buffer and run `eval-buffer`, which should spin up a second Emacs session in a self-contained temporary location.)

```emacs-lisp
;; -*- lexical-binding: t; -*-

(elpaca-test
  :interactive t
  :early-init
  (setq byte-compile-warnings '(not obsolete))
  (setq inhibit-startup-screen t)
  :init
  (load-theme 'modus-vivendi :no-confirm)
  (elpaca jinx)
  (elpaca elixir-mode)
  (elpaca-wait)

  (let ((buff (generate-new-buffer "Bug example")))
    (set-buffer buff)
    (elixir-mode)
    (jinx-mode)
    (insert "# this is an elixir comment that shouldn't have a spelling error here")
    (display-buffer buff))

  (use-package jinx
    :defer t
    :hook ((text-mode . jinx-mode)
           (prog-mode . jinx-mode))
    :bind (("C-;" . jinx-correct))
    :custom
    (jinx-camel-modes '(prog-mode))
    (jinx-delay 0.1)))
```

You can see in the `elixir-mode` buffer that the comment has a misspelling marked under `shouldn` in `shouldn't`. Any ideas where I could look to fix this?

**Comment by minad** [comment]
I suspect that elixir-mode modifies the syntax table in some way, which changes the treatment of the single quote. However Jinx installs its own syntax table during Jinx operation, see jinx--syntax-table, so this shouldn't be a problem in principle. Maybe Jinx somewhere misses to use jinx--syntax-table somewhere, where it actually should.

**Reply by yalfg** [reply]
It seems related to supporting multiple languages. In my case I had a similar issue with the hyphen, which happened after I selected two languages in the Jinx configuration (en_US and fr_FR). Then a word as "set-up" is shown as an error when inside a C comment for example. If I set the languages to "en_US" only, no problem.

I looked a bit at how word detection is done, and some enchant doc indicated it was using the English rules. But maybe that is obsolete, because I don't get the different behavior when adding French then. Still, it seems related to multiple languages support, and I haven't dug further yet. Any suggestion welcome.

**Reply by minad** [reply]
I don't see this problem in C files. Only tyxpo is highlighted as a typo.

~~~ c
/* set-up tyxpo don't maison */
/* Local Variables: */
/* jinx-languages: "fr en" */
/* End: */
~~~

**Reply by yalfg** [reply]
Very weird: no problem on my side either with "fr en", or "en fr". But with "en_US fr_FR" I reliably have the issue. Could you try? In any case I now have a work-around ;)

**Reply by yalfg** [reply]
I tried more combinations, and it's the en_US that triggers the issue in my case. Alone, it's OK. With another language, the hyphen becomes a problem. "en" only is fine. "en_GB" also has the issue. So somehow, it seems specializing the English variant in combination with another language (fr in my case) leads to a change in word splitting.

**Reply by yalfg** [reply]
By the way, this is with Debian Bookworm. Enchant, hunspell and aspell are installed (both with en and fr). Now Enchant is configured to prioritize aspell for en_US, and otherwise hunspell. So hunspell would be prioritized for fr_FR and also en. I notice that "en_US fr", the combination creating an issue for me, combines different priorities (en_US => aspell, fr =>  hunspell). Could this leads to this strange side effect?... No idea, could be a coincidence.

**Reply by yalfg** [reply]
I created a simple `$HOME/.config/enchant/enchant.ordering` prioritizing hunspell in all cases, it made no difference. I still have the hyphen issue with "en_US fr", and not with "en fr", even when en_US should also pick hunspell. So the priority difference doesn't seem to play.

**Reply by bestlem** [reply]
I see this hyphen issue as well in org-mode using en_GB and enchant using Applespell
